server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -189,6 +192,7 @@ public IndexRequest(@Nullable ShardId shardId, StreamInput in) throws IOExceptio
                     : new ArrayList<>(possiblyImmutableExecutedPipelines);
             }
         }
+        isFieldInferenceDone = in.getTransportVersion().before(SEMANTIC_TEXT_FIELD_ADDED) || in.readBoolean();
```

Comment:
Generally, if we are serializing from an older version the default is `false`. Why are you defaulting to `true` here?

Reply:
The variable register whether the field inference has already been done, that is, it's not necessary. So previous versions have that as true.

I can change the variable name (and meaning) to `needsFieldInference` or similar, if that makes it clearer and in line with other features 👍